keras_cv/bounding_box/utils.py

```diff
@@ -87,3 +87,110 @@ def _format_outputs(boxes, squeeze):
     if squeeze:
         return tf.squeeze(boxes, axis=0)
     return boxes
+
+
+def pad_with_sentinels(bounding_boxes):
+    """Pads the given bounding box tensor with -1s.
+
+    This is done to convert RaggedTensors to be converted into
+    standard Dense tensors, which have better performance and
+    compatibility within the TensorFlow ecosystem.
+
+    Args:
+        bounding_boxes: a ragged tensor of bounding boxes.
+        Can be batched or unbatched.
+
+    Returns:
+        a Tensor containing the -1 padded bounding boxes.
+    """
+    return bounding_boxes.to_tensor(-1)
+
+
+def filter_sentinels(bounding_boxes):
+    """converts a Dense padded bounding box `Tensor` to a `tf.RaggedTensor`.
+
+    Args:
+        bounding_boxes: a Tensor of bounding boxes.  May be batched, or unbatched.
+
+    Returns:
+        `tf.RaggedTensor` containing the filtered bounding boxes.
+    """
+    is_ragged = isinstance(bounding_boxes, tf.RaggedTensor)
+    is_batched = len(bounding_boxes.shape) == 3
+    if is_batched:
+        if is_ragged:
+            bounding_boxes = bounding_boxes.to_tensor(-1)
+        mask = bounding_boxes[:, :, 4] != -1
+        filtered_bounding_boxes = tf.ragged.boolean_mask(bounding_boxes, mask)
+    else:
+        mask = bounding_boxes[:, 4] != -1
+        filtered_bounding_boxes = tf.boolean_mask(bounding_boxes, mask)
+        filtered_bounding_boxes = tf.RaggedTensor.from_tensor(filtered_bounding_boxes)
+    return filtered_bounding_boxes
+
+
+def add_class_id(bounding_boxes, class_id=0):
+    """Add class ID to the innermost Tensor or RaggedTensor representing bounding boxes.
+
+    Bounding box utilities in Keras_CV expects bounding boxes to have class IDs
+    along with bounding box cordinates. This utility adds a class ID to the
+    innermost tensor representing the bounding boxes.
+
+    Usage:
+    ```python
+    bounding_boxes = tf.random.uniform(shape=[2, 2, 4])
+    bounding_boxes_with_class_id = keras_cv.bounding_box.add_class_id(
+                                    bounding_boxes, class_id=1)
+    bounding_boxes_with_class_id is a Tensor of shape [2, 2, 5]
+    ```
+
+    Args:
+        bounding_boxes: a `tf.Tensor` of bounding_boxes, may be batched unbatched.
+        class_id: (Optional) The value of class id that needs to be padded.
+            Defaults to 0.
+
+    Returns:
+        `tf.Tensor` with an additional class id padded to the original bounding boxes.
+    """
+    # format input bounding boxes
+    is_ragged = isinstance(bounding_boxes, tf.RaggedTensor)
+
+    if is_ragged:
+        row_lengths = list(bounding_boxes.nested_row_lengths())
+        # increase row length to account for clas-id addition
+        row_lengths[1] = row_lengths[1] + 1
+        bounding_boxes = bounding_boxes.to_tensor()
+
+    # pad input bounding boxes
+    if bounding_boxes.shape[-1] > 4:
```

Comment:
any reason to not check that it is equal to 4 here?  > doesn't catch cases of [0, 1, 2 ,3]

Reply:
I checked with your regarding this. The bounding_boxes could get more than 5 values.